api/src/Subscriber/RegistrationSubscriber.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Subscriber;
+
+use ApiPlatform\Symfony\EventListener\EventPriorities;
+use App\Entity\User;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use JetBrains\PhpStorm\ArrayShape;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+use Twig\Environment;
+
+class RegistrationSubscriber implements EventSubscriberInterface
+{
+    private EntityManagerInterface $entityManager;
+    private MailerInterface $mailer;
+    private Environment $twig;
+
+    public function __construct(EntityManagerInterface $entityManager, MailerInterface $mailer, Environment $twig)
+    {
+        $this->entityManager = $entityManager;
+        $this->mailer = $mailer;
+        $this->twig = $twig;
+    }
+
+    #[ArrayShape([KernelEvents::VIEW => "array"])]
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::VIEW => ['handleRegistration', EventPriorities::PRE_WRITE]
+        ];
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     * @throws Exception
+     */
+    public function handleRegistration(ViewEvent $event): void
+    {
+        /** @var $user User */
+        $user = $event->getControllerResult();
+
+        if (!($user instanceof User && Request::METHOD_POST === $event->getRequest()->getMethod())) {
+            return;
+        }
+
+        $token = hash('sha512', $user->getEmail().(new \DateTime())->format('Y-m-d H:i:s'));
+
+        $user->setToken($token);
+
+        $this->entityManager->flush();
```

Comment:
Group them to improve readability.

Reply:
I cant use the hash in the setToken because it is used after in $activationURL